src/move/econia/sources/market.move

```diff
@@ -2839,10 +2842,16 @@ module econia::market {
                     // If order at head of the queue is in price level:
                     if (option::contains(
                             &avl_queue::get_head_key(avlq_ref_mut), &price)) {
-                        // Pop order, storing only its size.
-                        let Order{size: order_size, price: _, user: _,
-                                  custodian_id: _, order_access_key: _} =
-                            avl_queue::pop_head(avlq_ref_mut);
+                        // Pop order, storing only its size and price.
+                        let Order{
+                            size: order_size,
+                            price: order_price,
+                            user: _,
+                            custodian_id: _,
+                            order_access_key: _
+                        } = avl_queue::pop_head(avlq_ref_mut);
+                        // Verify order price equals insertion key.
+                        assert!(order_price == price, E_ORDER_PRICE_MISMATCH);
```

Comment:
Why add this here? Seems like something that should/would go everywhere we call `pop_head` (or even anywhere we get an order at all). Basically what makes this location special.

Reply:
First I'll note that the codebase doesn't have any runtime logic for mutating an order's price field once it has been submitted

The other code already was already evaluated as not a security risk, and this only offered "as a small suggestion" from an auditor on new code

Reply:
Okay, i see.